src/hooks/Tokens.ts

```diff
@@ -181,8 +182,16 @@ export function useCurrency(currencyId: string | null | undefined): Currency | n
   const { chainId } = useActiveWeb3React()
   const isETH = currencyId?.toUpperCase() === 'ETH'
   const token = useToken(isETH ? undefined : currencyId)
-  const extendedEther = useMemo(() => (chainId ? ExtendedEther.onChain(chainId) : undefined), [chainId])
+  const extendedEther = useMemo(
+    () =>
+      chainId
+        ? ExtendedEther.onChain(chainId)
+        : // display mainnet when not connected
+          ExtendedEther.onChain(SupportedChainId.MAINNET),
```

Comment:
i really don't like this default chain id here (i think this is the only place we do this), what needs to change is the app needs to control the selected network instead of the connector

Reply:
That should be a larger change in the future. I think it will also help with first paint and linking semantics (ie links can rely on the chain being correct).